Examcc/src/exo2/SortedContainer.java

```diff
@@ -0,0 +1,35 @@
+package exo2;
+
+import java.lang.reflect.Array;
+import java.util.Arrays;
+
+public class SortedContainer implements ISortable {
+	
+	int[] element;
+	int listSize;
+	
+	public SortedContainer(int[] list){
+		element = list;
+		this.listSize = list.length;
+	}
+	
+	public void sort(int[] list,int listSize) {
+		element = list;
+		Arrays.sort(list);
+	}
+	
+	public int[] elements() {
+		return element;
+	}
+	public void add(int value) {
+		int [] newelem;
+		newelem=new int[(element.length)+1];
+		for (int i=0;i<(element.length); i++) {
+			newelem[i] = element[i];
+		}
+		newelem[newelem.length-1]=value;
+		sort(newelem,newelem.length);
```

Comment:
Ok, this works but... Unfortunately this is totally ineffecient and resource consuming!
Each time you add an element, you make a copy of the previous list!
What if your initial list contains 1000000 elements?